connector_ecommerce/product.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Sébastien BEAU
+#    Copyright 2011-2013 Akretion
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+from openerp.addons.connector.session import ConnectorSession
+from .event import on_product_price_changed
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    # TODO implement set function and also support multi tax
+    @api.depends('taxes_id', 'taxes_id.group_id')
+    def _get_tax_group_id(self):
+        for template in self:
+            taxes = template.taxes_id
+            template.tax_group_id = taxes[0].group_id.id if taxes else False
+
+    tax_group_id = fields.Many2one(
+        comodel_name='account.tax.group',
+        compute='_get_tax_group_id',
+        string='Tax Group',
+        help='Tax group are used with some external '
+             'system like Prestashop',
+    )
+
+    @api.multi
+    def _price_changed(self, vals):
+        """ Fire the ``on_product_price_changed`` on all the variants of
+        the template if the price if the product could have changed.
+
+        If one of the field used in a sale pricelist item has been
+        modified, we consider that the price could have changed.
+
+        There is no guarantee that's the price actually changed,
+        because it depends on the pricelists.
+        """
+        type_model = self.env['product.price.type']
+        price_fields = type_model.sale_price_fields()
+        # restrict the fields to the template ones only, so if
+        # the write has been done on product.product, we won't
+        # update all the variants if a price field of the
+        # variant has been changed
+        tmpl_fields = [field for field in vals if field in self._fields]
+        if any(field in price_fields for field in tmpl_fields):
+            product_model = self.env['product.product']
+            session = ConnectorSession(self.env.cr, self.env.uid,
+                                       context=self.env.context)
+            products = product_model.search(
+                [('product_tmpl_id', 'in', self.ids)]
```

Comment:
You are looking on product model, but self.ids are product.template records, isn't it?